### PR TITLE
Dialogue updates

### PR DESCRIPTION
**Pull request by prepsure**
- Dialogue is now determined by a single file on an interactable that defines the entire interaction
- Entire new spec defined and implemented
- Old dialogue handler is now defunct
- Support for saving and loading flags in the dialogue, to keep track of conversations
- Needs to be tested once integrated with the new save/load system

- Parser soon to come

### CODE REVIEW

**Comment by rping-qwfpgj** [comment]
Looks fine, just merge main into the branch to resolve conflicts

File: Assets/Scripts/DialogueSystem/Characters.cs [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
-namespace Assets.Scripts.Dialogue_System
+namespace Assets.Scripts.DialogueSystem
 {
     static class Characters
     {
         public static NarrativeCharacter OverheadSpeaker =
             new("Overhead Speaker", new Color(255, 255, 255));
         public static NarrativeCharacter Opus = new("Opus", new Color(255, 255, 255));
         public static NarrativeCharacter Coda = new("Coda", new Color(255, 255, 0));
+        public static NarrativeCharacter Clef = new("Coda", new Color(0, 255, 0));
```
Comment (cadenceblorbo) [inline Assets/Scripts/DialogueSystem/Characters.cs:16]: should be "clef" probably

File: Assets/Scripts/DialogueSystem/DialogueSamples/ExampleInput.cs
Comment (cadenceblorbo) [inline Assets/Scripts/DialogueSystem/DialogueSamples/ExampleInput.cs:1]: commented out, is this still needed?

**Review by cadenceblorbo — COMMENTED** [review]
Looks pretty solid! Not as thorough a review on the code side of things bc timebox soon.
The only bugs I found in testing in the editor are I think caused by the old pause state.

**Comment by cadenceblorbo** [comment]
As Paige said, probably a good area to revisit for spring cleaning.

**Pull request merged**